src/components/CurriculumComponents/UnitsTab/UnitsTab.tsx

```diff
@@ -238,11 +258,21 @@ const UnitsTab: FC<UnitsTabProps> = ({ data, examboardSlug }) => {
         });
       }
       filters.tiers.sort((a, b) => a.tier_slug.localeCompare(b.tier_slug));
+      // Sort disciplines
+      filters.disciplines.sort((a, b) => a.title.localeCompare(b.title));
+
+      // Add an "All" option if there are 2 or more disciplines. Set to -1 id as this shouldn't ever appear in the DB
+      const allDisciplineTag: Discipline = { id: -1, title: "All" };
+      if (filters.disciplines.length >= 2) {
+        filters.disciplines.unshift(allDisciplineTag);
+      }
+
       initialYearSelection[year] = {
         subject:
           filters.childSubjects.find(
             (s) => s.subject_slug === "combined-science",
           ) ?? null,
+        discipline: allDisciplineTag,
```

Comment:
Took a slightly different approach to the existing pattern. The selected discipline for the year will initially be 'all' (-1), even if there are no disciplines. In the isVisibleUnit function (which looks after the filtering of visible units) will return true for disciplines if the value is 'all' (-1).